### PR TITLE
strategy: rebalance on KLine closed

### PR DESCRIPTION
**Pull request by narumiruna**
- Rebalance on KLine closed
- Add dry-run option

### CODE REVIEW

**Comment by austin362667** [comment]
can it run with multiple assets?

**Comment by narumiruna** [comment]
> can it run with multiple assets?

Yes, you can define the target weight of each asset in the config file:
https://github.com/c9s/bbgo/blob/3713c00c72d16f441f5fad91304269c5c4cd1689/config/rebalance.yaml#L13

**Comment by austin362667** [comment]
great! thx

File: pkg/strategy/rebalance/strategy.go [hunk 1]
```diff
@@ -94,26 +88,17 @@ func (s *Strategy) Validate() error {
...
+func (s *Strategy) Subscribe(session *bbgo.ExchangeSession) {
+	for _, symbol := range s.getSymbols() {
+		session.Subscribe(types.KLineChannel, symbol, types.SubscribeOptions{Interval: s.Interval.String()})
+	}
+}
 
 func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, session *bbgo.ExchangeSession) error {
 	s.TargetWeights = Normalize(s.TargetWeights)
-
-	go func() {
```
Comment (c9s) [inline pkg/strategy/rebalance/strategy.go:102]: @austin362667 since this was a go routine, this can not be back-tested.

**Review by c9s — APPROVED** [review]

**Pull request merged**